MediaAdminBundle/Controller/Api/MediaController.php

```diff
@@ -89,4 +89,36 @@ public function deleteAction($mediaId)
 
         return array();
     }
+
+    /**
+     * @param Request $request
+     * @param string  $folderId
+     *
+     * @Config\Route("/upload/{folderId}", name="open_orchestra_api_media_upload")
+     * 
+     * @return Response
+     */
+    public function uploadAction($folderId, Request $request)
+    {
+        $uploadedFile = $request->files->get('file');
+
+        if ($uploadedFile) {
+            $folderRepository = $this->get('open_orchestra_media.repository.media_folder');
+            $folder = $folderRepository->find($folderId);
+
+            $mediaClass = $this->container->getParameter('open_orchestra_media.document.media.class');
+            $media = new $mediaClass();
+            $media->setMediaFolder($folder);
+            $media->setFile($uploadedFile);
+
+            $documentManager = $this->get('object_manager');
+            $documentManager->persist($media);
+
+            if ($documentManager->flush()) {
+                $this->get('session')->getFlashBag()->add('success', $this->get('translator')->trans('open_orchestra_media_admin.form.media.success'));
+            }
+        }
+
+        return array();
```

Comment:
Instead of return nothing, you should return a 201 response code.

You could also return the media facade, this way you could analyse the datas to display the preview